### ISSUE TITLE
Chunky Encoding?

### ISSUE THREAD

**Issue by greggman** [beatgammit/base64-js#18]
I was trying to encode some relatively large audio data (32meg?). Calling `uint8ToBase64` crashed my browser. I changed the code to this

```
       var parts = [];

        // go through the array every three bytes, we'll deal with trailing stuff later
        for (i = 0, length = uint8.length - extraBytes; i < length; i += 3) {
          temp = (uint8[i] << 16) + (uint8[i + 1] << 8) + (uint8[i + 2])
          output += tripletToBase64(temp)
          if (output.length > 16384) {
            parts.push(output)
            output = ''
          }
        }
```

...

```
        parts.push(output);

        return parts.join('');
```

What do you think? I don't know what a good chunk size is but I'm sure you know each time you do `output +=` something JavaScript is probably allocating a string, copying the old string into it followed by whatever your appending. That means it should get progressively slower as output gets longer.

In any case I just thought I'd pass this along of it helps others. It doesn't make a whole lot of sense to me that Chrome would crash from this. I'd just expect a GC hiccup at worst. On the other hand the giant output string ends up allocating lots of giant strings toward the end whereas chunkifying it means only the last join make a giant string. I suppose this is voodoo code.

**Comment by feross** [comment]
Can you send a PR with a commit that adds a failing test case, then another commit with the changes you described that make the test start passing?

That would help make this more concrete.

**Comment by feross** [comment]
Released as 1.0.3.
